tfhe/src/integer/server_key/radix/slice.rs

```diff
@@ -0,0 +1,533 @@
+use std::ops::{Bound, RangeBounds};
+
+use crate::error::InvalidRangeError;
+use crate::integer::{RadixCiphertext, ServerKey};
+use crate::prelude::{CastFrom, CastInto};
+use crate::shortint;
+
+/// Normalize a rust bound object, and check that it is valid for the source integer
+pub(crate) fn parse_bounds<B, R>(
+    range: &R,
+    nb_bits: usize,
+) -> Result<(usize, usize), InvalidRangeError>
+where
+    R: RangeBounds<B>,
+    B: CastFrom<usize> + CastInto<usize> + Copy,
+{
+    let start = match range.start_bound() {
+        Bound::Included(inc) => (*inc).cast_into(),
+        Bound::Excluded(excl) => (*excl).cast_into() - 1,
```

Comment:
Shouldn't this be +1 here ?

If my range starts with `]x`, then the first index i include is x+1, not x - 1

Reply:
yes indeed !